Jetcaster/wear/src/main/java/com/example/jetcaster/ui/library/PodcastsScreen.kt

```diff
@@ -0,0 +1,206 @@
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.wear.compose.foundation.lazy.rememberScalingLazyListState
+import androidx.wear.compose.material.Button
+import androidx.wear.compose.material.ButtonDefaults
+import androidx.wear.compose.material.ChipDefaults
+import androidx.wear.compose.material.Icon
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.material.dialog.Alert
+import androidx.wear.compose.material.dialog.Dialog
+import com.example.jetcaster.R
+import com.example.jetcaster.core.data.model.PodcastInfo
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.composables.PlaceholderChip
+import com.google.android.horologist.composables.Section
+import com.google.android.horologist.composables.SectionedList
+import com.google.android.horologist.compose.layout.ScalingLazyColumnState
+import com.google.android.horologist.compose.layout.rememberColumnState
+import com.google.android.horologist.compose.material.Chip
+import com.google.android.horologist.compose.material.Title
+import com.google.android.horologist.images.coil.CoilPaintable
+
+@Composable
+fun PodcastsScreen(
+    podcastViewModel: PodcastViewModel = hiltViewModel(),
+    onPodcastsItemClick: (PodcastInfo) -> Unit,
+    onErrorDialogCancelClick: () -> Unit,
+) {
+    val uiState by podcastViewModel.uiState.collectAsStateWithLifecycle()
+
+    val modifiedState = when (uiState) {
+        is PodcastsScreenState.Loaded -> {
+            val modifiedPodcast = (uiState as PodcastsScreenState.Loaded).podcastList.map {
+                it.takeIf { it.title.isNotEmpty() }
+                    ?: it.copy(title = stringResource(id = R.string.no_title))
+            }
+
+            PodcastsScreenState.Loaded(modifiedPodcast)
+        }
+
+        PodcastsScreenState.Failed,
+        PodcastsScreenState.Loading,
+        -> uiState
+    }
+
+    PodcastsScreen(
+        podcastsScreenState = modifiedState,
+        onPodcastItemClick = {
+            onPodcastsItemClick(it)
+        },
+    )
+
+    if (modifiedState == PodcastsScreenState.Failed) {
+        Dialog(
+            showDialog = true,
+            onDismissRequest = onErrorDialogCancelClick,
+            scrollState = rememberScalingLazyListState(),
+        ) {
+            Alert(
+                title = {
+                    Text(
+                        text = stringResource(R.string.podcasts_no_podcasts),
+                        color = MaterialTheme.colors.onBackground,
+                        textAlign = TextAlign.Center,
+                        style = MaterialTheme.typography.title3,
+                    )
+                },
+            ) {
+                item {
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically,
+                    ) {
+                        Button(
```

Comment:
TIOLI: com.google.android.horologist.compose.material.Button cleans this up a lot.

Reply:
thanks (I'll need to fix it in the media toolkit sample as well)